### DISCUSSION TITLE
Reassign Content dialog is unclear / overlooked

### DISCUSSION THREAD

**Discussion by thisisjamessmith** [craftcms/cms#8756] [Ideas]
### Description

Over the years in both Craft and EE, my clients always seem to ignore or mis-interpret the option that appears when deleting a user asking them if they'd like to reassign the user's content to someone else or delete it (I suspect from the author's point of view, if they're not paying 100% attention and don't actually read the message, they instinctively think that it's just another confirmation button asking if they're sure they want to proceed with deleting the user.

### Additional info

- Craft version: 3.3.16.1

**Comment by thisisjamessmith** [comment]
The UX in this area is exacerbated by the fact that the exact same 'reassign content' modal window appears even when deleting users who have no content, so for an editor who is going through methodically deleting old user accounts, they will become accustomed to always choosing the 'delete' option, even though once in a while they'll come across a back-end CP user who does have some content which should be reassigned.

My usual recommendation to clients is to never delete back-end users and instead always keep them in a suspended status so that an accurate history of entry authorship is maintained, however, you can understand why for regulatory reasons some industries may be required to 'hard delete' inactive users.

**Comment by thisisjamessmith** [comment]
If you're deleting multiple users en-masse, the modal window doesn't break down which users have content attached to them, either.

**Comment by brandonkelly** [comment]
_Each GitHub issue should only focus on a single thing, and considering the UX feedback has taken over in your follow-up comments, I’ve edited the OP to focus on that._

**Comment by brandonkelly** [comment]
Part of the reason we added soft-delete support in Craft 3.1 was for this issue. At least there is a way to “undo” the user/entry deletion now. But I agree it could be more clear what exactly is going to be deleted/transferred.

This is also related to commerce#1048, where users that have had any Commerce subscriptions aren’t able to be deleted at all.

We are thinking we will tackle both of them in Craft 4 by replacing the current delete confirmation dialog with a modal that lists any issues that are preventing the user from being deleted, with inline quick actions for resolving them, one-by-one. If a user has any entries, the quick action will be to view those entries, and from there, you’ll be able to batch-reassign them to someone else.

**Comment by lindseydiloreto** [comment]
We just got bit pretty badly by this. I thought I'd seen a similar thread with a lot more upvotes, but here is our sob story...

One of my smartest clients, who has been working with the website for years, has apparently been selecting "Delete it" because the messaging was so unclear. She did not realize that "entries" implied all of the related data that our users had been creating. The site contains tens of thousands of entries, so we don't know if anything has been deleted permanently over the years. 😕

I humbly propose this text change...

**CURRENT**
<img width="337" alt="content@2x" src="https://user-images.githubusercontent.com/5309692/99456022-92295d80-28dd-11eb-84c5-d08382ec7870.png">

**PROPOSED**
<img width="340" alt="proposed@2x" src="https://user-images.githubusercontent.com/5309692/99456012-8c337c80-28dd-11eb-8b7e-e36d5fb1290c.png">

**Comment by brandonkelly** [comment]
I’m not convinced extrapolating what content will be deleted would make a significant impact here. Either way, had they read the message, they would know what they’re doing. The problem is they’re not reading the message to begin with.

I suspect the reason people skip over that text is that they assume that this is _just_ a typical confirmation dialog, which is trying to be scary by making you confirm you really want to delete the user multiple times – once by clicking the “Delete it” radio (where you may assume “it” just refers to the user, having not read anything else in the dialog), and a second time by clicking the submit button.

I just changed that radio button’s label to “Delete **their content**” for the next release. If I’m right, then that could improve things quite a bit.

<img width="405" alt="The new user deletion confirmation dialog, which now has a option labelled “Delete their content” rather than “Delete it”." src="https://user-images.githubusercontent.com/47792/99466748-ca3a9b80-28f1-11eb-9b45-2047bcb36259.png">

**Comment by lindseydiloreto** [comment]
Maybe, but that mostly seems like a lateral move to me. "it" vs "their content" is practically the same generic language.

I fudged the screenshot, we aren't using "Articles" and "Recommendations". We are using "Films" and "Companies". Each user is tied to a single Company, and multiple Films.

My client is smart. If she had seen the actual section names in there, I'm almost 100% certain it would have stopped her. She would have stopped in her tracks if she saw a message talking about deleting **1 Company** and **41 Films**.

Yes, they are all "entries" under the hood. But that doesn't mean much to the client, unfortunately.

**Comment by brandonkelly** [comment]
Guess it wouldn’t hurt to do both. Just made that change for the next release as well.

<img width="407" alt="A user deletion confirmation dialog" src="https://user-images.githubusercontent.com/47792/99573654-36b5a900-298b-11eb-8c7a-2cc0292f8074.png">

**Comment by lindseydiloreto** [comment]
Ah, brilliant! Thanks @brandonkelly, greatly appreciated! 🍺

It's hard to quantify, but I'm fairly confident this will prevent at least a few avoidable tragedies. 😄

**Comment by proimage** [comment]
As a further minor enhancement to this, I find it's always useful to have the explicit action described in the button itself.

In this case, if they've selected "Delete their content", then the button could read "**Delete users & their content**" (or "**Delete users & content**" for brevity), and dynamically switch to just "Delete users" if the "Transfer" option is selected.

**Comment by brandonkelly** [comment]
@proimage Good idea, just made that change as well.
